library/Icinga/Web/RememberMe.php

```diff
@@ -0,0 +1,188 @@
+<?php
+/* Icinga Web 2 | (c) 2020 Icinga Development Team | GPLv2+ */
+
+namespace Icinga\Web;
+
+use Icinga\Application\Config;
+use Icinga\Authentication\Auth;
+use Icinga\Crypt\RSA;
+use Icinga\Rememberme\Common\Database;
+use Icinga\User;
+use ipl\Sql\Select;
+use UnexpectedValueException;
+
+class RememberMe
+{
+    use Database;
+
+    /**
+     * Constant cookie
+     */
+    const COOKIE = 'remember-me';
+
+    /**
+     * @var string
+     */
+    protected $encryptedPassword;
+
+    /**
+     * @var string
+     */
+    protected $username;
+
+    /**
+     * @var object
+     */
+    protected $rsa;
+
+    /**
+     * Check if cookie is set
+     *
+     * @return bool
+     *
+     */
+    public static function hasCookie()
+    {
+        return isset($_COOKIE[static::COOKIE]);
+    }
+
+    /**
+     * Get cookie values
+     *
+     * @return void|static
+     */
+    public static function fromCookie()
+    {
+        $data = explode('|', $_COOKIE[static::COOKIE]);
+        $publicKey = base64_decode(array_pop($data));
+
+        $select = (new Select())
+            ->from('rememberme')
+            ->columns('*')
+            ->where(['public_key = ?' => $publicKey]);
+
+        $rememberMe = new static();
+        $dbData = $rememberMe->getDb()->select($select)->fetch();
+        $newData = [];
+        foreach ($dbData as $key => $value) {
+            $newData[$key] = $value;
+        }
+
+        $rememberMe->rsa = (new RSA())->loadKey($newData['private_key'], $publicKey);
+        $rememberMe->username = $rememberMe->rsa->decryptFromBase64($data[0]);
+        $rememberMe->encryptedPassword = $data[1];
+
+        return $rememberMe;
+    }
+
+    /**
+     * Encrypt the given username, password and assign the variables
+     *
+     * @param $username
+     * @param $password
+     *
+     * @return static
+     */
+    public static function fromCredentials($username, $password)
+    {
+
+        $rememberMe = new static();
+
+        $rsa = (new RSA())->loadKey(...RSA::keygen());
+
+        $rememberMe->encryptedPassword = $rsa->encryptToBase64($password);
+        $rememberMe->username = $username;
+        $rememberMe->rsa = $rsa;
+
+        return $rememberMe;
+    }
+
+    /**
+     * Set the values for 'remember-me' cookie
+     *
+     * @return Cookie
+     */
+    public function getCookie()
+    {
+        $value = $this->rsa->encryptToBase64($this->username);
+        $value[] = $this->encryptedPassword[0];
```

Comment:
Watch out here. `$encryptedPassword` must never be an array but a string. And array access also works for strings.